src/Components/Facility/FacilityDoctorList.tsx

```diff
@@ -0,0 +1,140 @@
+import useQuery from "../../Utils/request/useQuery";
+import routes from "../../Redux/api";
+import { DoctorModal } from "./models";
+import DoctorsCountCard from "./DoctorsCountCard";
+import { DoctorIcon } from "../TeleIcu/Icons/DoctorIcon";
+
+export const FacilityDoctorList = (props: any) => {
+  const [doctorCapacityModalOpen, setDoctorCapacityModalOpen] = useState(false);
+  const [totalDoctors, setTotalDoctors] = useState(0);
+
+  const doctorQuery = useQuery(routes.listDoctor, {
+    pathParams: { facilityId: props.facilityId },
+    onResponse: ({ res, data }) => {
+      if (res?.ok && data) {
+        let totalCount = 0;
+        [data].map((doctor: DoctorModal) => {
+          if (doctor.count) {
+            totalCount += doctor.count;
+          }
+        });
+        setTotalDoctors(totalCount);
+      }
+    },
+  });
+
+  let doctorList: any = null;
+  if (!doctorQuery.data || !doctorQuery.data.results.length) {
+    doctorList = (
+      <h5 className="flex w-full items-center justify-center rounded-lg bg-white p-4 text-xl font-bold text-gray-500 shadow">
+        No Doctors Found
+      </h5>
+    );
+  } else {
+    doctorList = (
+      <div className="mt-4 grid gap-6 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4">
+        {/* Total Doctors Count Card */}
+        <div className="w-full">
+          <div className="flex h-full flex-col rounded-sm border border-primary-500 bg-primary-100 shadow-sm">
+            <div className="flex flex-1 items-center justify-start gap-3 px-4 py-6">
+              <div className="rounded-full bg-primary-500 p-4">
+                <DoctorIcon className="h-5 w-5 fill-current text-white" />
+              </div>
+              <div>
+                <div className="text-sm font-medium text-[#808080]">
+                  Total Doctors
+                </div>
+                <h2 className="mt-2 text-xl font-bold">{totalDoctors}</h2>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        {doctorQuery.data.results.map((data: DoctorModal) => {
+          const removeCurrentDoctorData = (doctorId: number | undefined) => {
+            if (doctorQuery.data !== undefined) {
+              doctorQuery.data.results = doctorQuery.data?.results.filter(
+                (i: DoctorModal) => i.id !== doctorId
+              );
+            }
+          };
+
+          return (
+            <DoctorsCountCard
+              facilityId={props.facilityId}
+              key={`bed_${data.id}`}
+              handleUpdate={async () => {
+                doctorQuery.refetch();
+                if (doctorQuery.res?.ok && doctorQuery.data) {
+                  // update total doctors count
+                  let totalCount = 0;
+                  doctorQuery.data.results.map((doctor: DoctorModal) => {
+                    if (doctor.count) {
+                      totalCount += doctor.count;
+                    }
+                  });
+                  setTotalDoctors(totalCount);
```

Comment:
Total count and all are "side-effects" of the list query. You need not compute it again upon delete, it should be computed automatically when `query.data.results` changes